Classes/Middleware/Frontend/LanguageModeSwitch.php

```diff
@@ -138,4 +155,27 @@ private function getNewLanguageWithMode(
             ])
         );
     }
+
+    private function pageHasStandAloneContent(int $pageId, int $languageId): bool
+    {
+        $queryBuilder = clone $this->queryBuilder;
```

Comment:
I'm reusing the same QueryBuilder for 2 requests for different tables.

That's not how you would do it by the book, because you would request a separate QueryBuilder per table.

We sometimes do it like this in our project to keep our code concise and for tables that are coupled together (there are several JOIN statements in the core over `pages` and `tt_content`) and therefore can never be on separate connections.

If you want it changed, I can inject a 2nd QueryBuilder.